src/routes/webhook/webhook-receiver-post.ts

```diff
@@ -0,0 +1,77 @@
+import { BinaryLike, createHmac } from "crypto";
+import { Request, Response } from "express";
+import { validationResult } from "express-validator";
+import { getLogger } from "~/src/config/logger";
+import { GitHubServerApp } from "~/src/models/git-hub-server-app";
+import { WebhookContext } from "./webhook-context";
+
+export const WebhookReceiverPost = async (request: Request, response: Response): Promise<void> => {
+	const errors = validationResult(request);
+	if (!errors.isEmpty()) {
+		response.status(400).json({ errors: errors.array() });
+		return;
+	}
+	const logger = getLogger("webhook.receiver");
+	const eventName = request.headers["x-github-event"] as string;
+	const signatureSHA256 = request.headers["x-hub-signature-256"] as string;
+	const id = request.headers["x-github-delivery"] as string;
+	const uuid = request.params.uuid;
+	let webhookSecret: string = process.env.WEBHOOK_SECRET!;
+	const payload = request.body;
+	try {
+		if (uuid != "cloud") {
+			const gitHubServerApp = await GitHubServerApp.findForUuid(uuid);
+			if (!gitHubServerApp) {
+				response.status(400).send("GitHub app not found");
+				return;
+			}
+			webhookSecret = gitHubServerApp?.webhookSecret;
+		}
+		const verification = createHash(JSON.stringify(payload), webhookSecret);
+		if (verification != signatureSHA256) {
+			response.status(400).send("signature does not match event payload and secret");
+			return;
+		}
+
+		const webhookContext = new WebhookContext({
```

Comment:
WebhookContext is still missing important github app information to be able to continue like the app url, clientId, webhooksecret, etc (pretty much everything in the GithubServerApp table).  You also need to a way to add the github.com app information in there if it's not a ghe app

Reply:
I'd also add the action to the webhook context

Reply:
I'll add missing information in `WebhookContext` in upcoming PRs

Reply:
action is now part of `WebhookContext`